auth/src/main/java/com/firebase/ui/auth/util/SmartLock.java

```diff
@@ -58,38 +60,13 @@
     private GoogleApiClient mCredentialsApiClient;
 
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        setContentView(R.layout.save_credentials_layout);
-
-        if (!FirebaseAuthWrapperFactory.getFirebaseAuthWrapper(mActivityHelper.getAppName())
-                .isPlayServicesAvailable(this)) {
-            finish(RESULT_FIRST_USER, getIntent());
-            return;
-        }
-
-        mName = getIntent().getStringExtra(ExtraConstants.EXTRA_NAME);
-        mEmail = getIntent().getStringExtra(ExtraConstants.EXTRA_EMAIL);
-        mPassword = getIntent().getStringExtra(ExtraConstants.EXTRA_PASSWORD);
-        mProvider = getIntent().getStringExtra(ExtraConstants.EXTRA_PROVIDER);
-        mProfilePictureUri = getIntent()
-                .getStringExtra(ExtraConstants.EXTRA_PROFILE_PICTURE_URI);
-
-        mCredentialsApiClient = new GoogleApiClient.Builder(this)
-                .addConnectionCallbacks(this)
-                .addOnConnectionFailedListener(this)
-                .addApi(Auth.CREDENTIALS_API)
-                .enableAutoManage(this, this)
-                .build();
-    }
-
-    @Override
     public void onConnected(@Nullable Bundle bundle) {
         if (mEmail == null) {
             Log.e(TAG, "Unable to save null credential!");
-            finish(RESULT_FIRST_USER, getIntent());
+            mActivity.finish(RESULT_OK, mActivity.getIntent());
```

Comment:
Yes. When this was an activity we finished with result `RESULT_FIRST_USER`, but the calling activity didn't actually care what the result was (all the calling activities finished with `RESULT_OK`). Since the fragment it closing the activity for it, we just finish everything with `RESULT_OK`.